lib/jekyll-github-metadata/edit-link-tag.rb

```diff
@@ -0,0 +1,102 @@
+module Jekyll
+  module GitHubMetadata
+    class EditLinkTag < Liquid::Tag
+      attr_reader :context
+
+      # Defines an instance method that delegates to a hash's key
+      #
+      # hash_method -  a symbol representing the instance method to delegate to. The
+      #                instance method should return a hash or respond to #[]
+      # key         - the key to call within the hash
+      # method      - (optional) the instance method the key should be aliased to.
+      #               If not specified, defaults to the hash key
+      # default     - (optional) value to return if value is nil (defaults to nil)
+      #
+      # Returns a symbol representing the instance method
+      def self.def_hash_delegator(hash_method, key, method, default = nil)
+        define_method(method) do
+          hash = send(hash_method)
```

Comment:
If we did, we'd have to make `site`, `site_github`, `source`, and `page` public.